tests/Integration/Behaviour/Features/Context/FeatureFlagFeatureContext.php

```diff
@@ -0,0 +1,159 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Integration\Behaviour\Features\Context;
+
+use Doctrine\ORM\EntityManager;
+use Exception;
+use PrestaShopBundle\Entity\FeatureFlag;
+use RuntimeException;
+
+class FeatureFlagFeatureContext extends AbstractPrestaShopFeatureContext
+{
+    /**
+     * "When" steps perform actions, and some of them store the latest result
+     * in this variable so that "Then" action can check its content
+     *
+     * @var mixed
+     */
+    protected $latestResult;
+
+    /**
+     * @Given /^I register a (disabled|enabled) feature flag "(.+)"$/
+     */
+    public function registerFlag(string $state, string $name): void
+    {
+        $doctrineEntityManager = $this->getDoctrineEntityManager();
+
+        try {
+            $newFeatureFlag = new FeatureFlag($name);
+
+            if ($state === 'enabled') {
+                $newFeatureFlag->enable();
+            }
+
+            $doctrineEntityManager->persist($newFeatureFlag);
+            $doctrineEntityManager->flush();
+        } catch (Exception $e) {
+            $this->latestResult = $e;
+        }
+    }
+
+    /**
+     * @When /^I (disable|enable) feature flag "(.+)"$/
+     */
+    public function enableFeatureFlag(string $state, string $name): void
+    {
+        $doctrineEntityManager = $this->getDoctrineEntityManager();
+
+        /** @var FeatureFlag $featureFlag */
+        $featureFlag = $doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findOneBy(['name' => $name]);
+
+        if ($state === 'enable') {
+            $featureFlag->enable();
+        } else {
+            $featureFlag->disable();
+        }
+
+        $doctrineEntityManager->flush();
+    }
+
+    /**
+     * @Then /^the feature flag "(.+)" state is (disabled|enabled)$/
+     */
+    public function assertFeatureFlagState(string $name, string $state): void
+    {
+        $doctrineEntityManager = $this->getDoctrineEntityManager();
+
+        /** @var FeatureFlag $featureFlag */
+        $featureFlag = $doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findOneBy(['name' => $name]);
+
+        if ($state === 'enabled' && !$featureFlag->getState()) {
+            throw new RuntimeException(sprintf('Feature flag %s is disabled although it was expected to be enabled', $name));
+        } elseif ($state === 'disabled' && $featureFlag->getState()) {
+            throw new RuntimeException(sprintf('Feature flag %s is enabled although it was expected to be disabled', $name));
+        }
+    }
+
+    /**
+     * @Then the system should reject my latest feature flag creation
+     */
+    public function featureFlagExcetion()
+    {
+        throw new PendingException();
+    }
+
```

Comment:
```suggestion

```

Reply:
Nice catch! Totally forgot this part. My bad.